src/Apps/Ingress/Models/Context/NavigationContext.tsx

```diff
@@ -0,0 +1,72 @@
+import produce from 'immer';
+import React, { useContext, useReducer } from 'react';
+import { getDebugLogger } from '../../../../Models/Services/Utils';
+import {
+    INavigationContext,
+    INavigationContextProviderProps,
+    INavigationContextState,
+    NavigationContextAction,
+    NavigationContextActionType,
+    PageNames
+} from './NavigationContext.types';
+
+export const NavigationContext = React.createContext<INavigationContext>(null);
+export const useNavigationContext = () => useContext(NavigationContext);
+
+const debugLogging = false;
+const logDebugConsole = getDebugLogger(
+    'Ingress - NavigationContext',
+    debugLogging
+);
+
+export const NavigationContextReducer: (
+    draft: INavigationContextState,
+    action: NavigationContextAction
+) => any = produce((draft, action) => {
+    logDebugConsole(
+        'info',
+        `Updating Navigation context ${action.type} with payload: `,
+        action.payload
+    );
+    switch (action.type) {
+        case NavigationContextActionType.NAVIGATE_TO:
+            draft.currentPage = action.payload.pageName;
+            break;
+    }
+});
+
+export const NavigationContextProvider: React.FC<INavigationContextProviderProps> = (
+    props
+) => {
+    const { children, initialState } = props;
+
+    const [navigationContextState, navigationContextDispatch] = useReducer(
+        NavigationContextReducer,
+        { ...emptyState, ...initialState },
```

Comment:
shouldn't `getInitialState` method handle this merge based on emptyState and initialState one line below? instead of spread merge here?

Reply:
I'm following the pattern @msnyder-msft was using for OatPageContext, don't remember why we needed the merge + the function here.